tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/es/BackupManagerElasticSearch.java

```diff
@@ -433,22 +430,23 @@ private List<SnapshotInfo> findSnapshots(final Long backupId) {
           String.format(
               "Encountered an error connecting to Elasticsearch while searching for snapshots. Repository name: [%s].",
               getRepositoryName());
-      throw new TasklistElasticsearchConnectionException(reason, ex);
+      throw new BackupRepositoryConnectionException(reason, ex);
     } catch (final Exception e) {
       if (isSnapshotMissingException(e)) {
         // no snapshot with given backupID exists
-        throw new NotFoundApiException(String.format("No backup with id [%s] found.", backupId), e);
+        throw new ResourceNotFoundException(
+            String.format("No backup with id [%s] found.", backupId), e);
       }
       if (isRepositoryMissingException(e)) {
         final String reason =
             String.format(
                 "No repository with name [%s] could be found.",
                 tasklistProperties.getBackup().getRepositoryName());
-        throw new TasklistRuntimeException(reason);
+        throw new GenericBackupException(reason);
       }
       final String reason =
           String.format("Exception occurred when searching for backup with ID [%s].", backupId);
-      throw new TasklistRuntimeException(reason, e);
+      throw new GenericBackupException(reason, e);
```

Comment:
:question: Why isn't this `ResourceNotFoundException`?

Reply:
:shrug:  I just mapped to its counterparty. This class will be deleted very soon